test/hotspot/gtest/nmt/test_nmt_buffer_overflow_detection.cpp

```diff
@@ -180,14 +180,162 @@ TEST_VM(NMT, test_realloc) {
   }
 }
 
-TEST_VM_FATAL_ERROR_MSG(NMT, memory_corruption_call_stack, ".*header canary.*") {
-  if (MemTracker::tracking_level() != NMT_detail) {
-    guarantee(false, "fake message ignore this - header canary");
+#else // ASAN is enabled
+
```

Comment:
Do we really need to duplicate this macro, just to give the test an own name? Certainly this can be simplified?

Reply:
Fixed.